docs/tutorials/blackjack_tutorial.py

```diff
@@ -0,0 +1,346 @@
+# ------------------------------
+# 
+
+def create_grids(agent, usable_ace=False):
+    
+    # convert our state-action values to state values 
+    # and build a policy dictionary that maps observations to actions
+    V = defaultdict(float)
+    policy = defaultdict(int)
+    for obs, action_values in agent.q_values.items():
+        V[obs] = np.max(action_values)
+        policy[obs] = np.argmax(action_values)
+    
+    X, Y = np.meshgrid(
+        np.arange(12,22), # players count
+        np.arange(1,11))  # dealers face-up card
+    
+    # create the value grid for plotting
+    Z = np.apply_along_axis(lambda obs: V[(obs[0], obs[1], usable_ace)], axis=2, arr=np.dstack([X, Y]))
+    value_grid = X, Y, Z
+    
+    # create the policy grid for plotting
+    policy_grid = np.apply_along_axis(lambda obs: policy[(obs[0], obs[1], usable_ace)], 
+                                      axis=2, arr=np.dstack([X, Y]))
+    return value_grid, policy_grid
+
+
+def create_plots(value_grid, policy_grid, title='N/A'):
+    
+    # create a new figure with 2 subplots (left: state values, right: policy)
+    X, Y, Z = value_grid
+    fig = plt.figure(figsize=plt.figaspect(0.4))
+    fig.suptitle(title, fontsize=16)
+
+    # plot the state values
+    ax1 = fig.add_subplot(1, 2, 1, projection='3d')
+    ax1.plot_surface(X, Y, Z, rstride=1, cstride=1,
+                    cmap='viridis', edgecolor='none')
+    plt.xticks(range(12, 22), range(12, 22))
+    plt.yticks(range(1,11), ['A'] + list(range(2,11)))
+    ax1.set_title('State values: ' + title);
+    ax1.set_xlabel("Player sum")
+    ax1.set_ylabel("Dealer showing")
+    ax1.zaxis.set_rotate_label(False)
+    ax1.set_zlabel('Value', fontsize=14, rotation=90)
+    ax1.view_init(20, 220)
+    
+    # plot the policy
+    a2x = fig.add_subplot(1, 2, 2)
+    ax2 = sns.heatmap(policy_grid, linewidth=0, annot=True, cmap="Accent_r", cbar=False)
+    ax2.set_title('Policy: ' + title);
+    ax2.set_xlabel("Player sum")
+    ax2.set_ylabel("Dealer showing")
+    ax2.set_xticklabels(range(12, 22))
+    ax2.set_yticklabels(['A'] + list(range(2,11)), fontsize=12)
+    
+    # add a legend
+    legend_elements = [Patch(facecolor='lightgreen', edgecolor='black', label='Hit'),
+                       Patch(facecolor='grey', edgecolor='black', label='Stick')]
+    ax2.legend(handles=legend_elements, bbox_to_anchor=(1.3, 1))
+    return fig
+
+
+# state values & policy with usable ace (ace counts as 11)
+value_grid, policy_grid = create_grids(agent, usable_ace=True)
+fig1 = create_plots(value_grid, policy_grid, title='With usable ace')
+plt.show()
+
+
+# state values & policy without usable ace (ace counts as 1)
+value_grid, policy_grid = create_grids(agent, usable_ace=False)
+fig2 = create_plots(value_grid, policy_grid, title='Without usable ace')
+plt.show()
+
+
+# %%
+# It's good practice to call env.close() at the end of your script,
+# so that any used ressources by the environment will be closed.
+
+env.close()
+
```

Comment:
Re-add the images with the plots. I don't see any advantage in executing this tutorial during build. you can revert the hyperparameters if you want.

Reply:
readded the plots as static images to the `/_static/img/tutorials/` folder and changed the hyperparameters back to a higher number of episodes ✅